components/datasets/form/steps/component.js

```diff
@@ -140,6 +123,26 @@ class Step1 extends React.Component {
     this.props.onChange({ subscribable });
   }
 
+  handleSortLayer = (layers) => {
+    const { dataDataset: { applicationConfig } } = this.state;
+    const { onChange } = this.props;
+
+    onChange({
+      applicationConfig: {
+        ...!applicationConfig && { // eslint-disable-line object-curly-newline
+          [process.env.APPLICATIONS]: { layerOrder: layers.map(_layer => _layer.id) }
+        },
+        ...applicationConfig && {
```

Comment:
Be careful if there's already an `applicationConfig` attribute. We cannot override others. We only update ours and never override other RW data here.